### PR TITLE
[LLC][Core] Move APIs supporting LLC from Experimental to Core

### LINKED ISSUE #24200: [LLC] [Core] Move Response.IsError() property into Core
[This PR](https://github.com/Azure/azure-sdk-for-net/pull/24122) added Response.IsError() as an extension method in Azure.Core.Experimental to prototype the design for [casting Responses to models](https://github.com/azure/azure-sdk-for-net/issues/23372).  This issue tracks the work to make `Response.Error` a property on the `Response` type in Azure.Core.

### LINKED ISSUE #24699: [LLC] Remove RequestPropertiesPolicy
@KrzysztofCwalina has raised a concern about putting fundamental pipeline logic into policies.  

This tracks the work to investigate alternatives (setting the ResponseClassifier and calling response.EvaluateError() in SendAsync), and consider the performance implications fo them and whether this is the correct approach to moving this functionality into Core.

### PR DESCRIPTION
**Pull request by annelo-msft**
- [x] Remove ClassifiedResponse from Experimental and adding a public IsError to Response  -- We will do this for RC1
- [x] Remove ResponsePropertiesPolicy from Experimental and making it an internal class in Core  -- We will modify the TransportPolicy to do this classification instead
- [x] Remove ResponseExtensions from Experimental and adding a constructor to RequestFailedException that takes Response  -- We will do this for RC1
- [x] Move LLC API tests to Core
- [x] Remove EXPERIMENTAL? <-- not as part of this change

Fixes https://github.com/azure/azure-sdk-for-net/issues/24200
Fixes https://github.com/azure/azure-sdk-for-net/issues/24699

In addition, this PR moves response initialization into the TransportPolicy.

### CODE REVIEW

**Review by ellismg — APPROVED** [review]

File: sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs [hunk 1]
```diff
@@ -88,6 +88,9 @@ void AddCustomerPolicies(HttpPipelinePosition position)
 
             var sanitizer = new HttpMessageSanitizer(diagnostics.LoggedHeaderNames.ToArray(), diagnostics.LoggedQueryParameters.ToArray());
 
+            responseClassifier ??= new ResponseClassifier();
+            responseClassifier.MessageSanitizer = sanitizer;
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs:92]: We shouldn't be mutating an instance that is passed in by the customer.

File: sdk/core/Azure.Core/src/Response.cs [hunk 2]
```diff
@@ -82,6 +82,18 @@ public virtual BinaryData Content
         /// </summary>
         public abstract void Dispose();
 
+        /// <summary>
+        /// </summary>
+        public bool IsError { get; internal set; }
+
+        internal ResponseClassifier? ResponseClassifier { get; set; }
+
+        internal void EvaluateError(HttpMessage message)
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/Response.cs:91]: Move this implementation into the HttpPipelineTransportPolicy?

File: sdk/core/Azure.Core/tests/HttpMessageTests.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        [Test]
+        public void TryGetPropertyReturnsFalseIfNotExist()
+        {
+            HttpMessage message = new HttpMessage(new MockRequest(), new ResponseClassifier());
+
+            Assert.False(message.TryGetProperty("someName", out _));
+        }
+
+        [Test]
+        public void TryGetPropertyReturnsValueIfSet()
```
Comment (pakrym) [inline sdk/core/Azure.Core/tests/HttpMessageTests.cs:20]: We have the same test in PipelineTests, why a copy?

File: sdk/core/Azure.Core/src/Response.cs [hunk 4]
```diff
@@ -82,6 +82,18 @@ public virtual BinaryData Content
         /// </summary>
         public abstract void Dispose();
 
+        /// <summary>
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/Response.cs:85]: Docs

File: sdk/core/Azure.Core/tests/HttpMessageTests.cs (hunk 3 above)
Comment (annelo-msft) [reply sdk/core/Azure.Core/tests/HttpMessageTests.cs:20]: I removed it from PipelineTests.

File: sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs [hunk 5]
```diff
@@ -88,6 +88,8 @@ void AddCustomerPolicies(HttpPipelinePosition position)
 
             var sanitizer = new HttpMessageSanitizer(diagnostics.LoggedHeaderNames.ToArray(), diagnostics.LoggedQueryParameters.ToArray());
 
+            responseClassifier ??= new ResponseClassifier() { MessageSanitizer = sanitizer };
```
Comment (KrzysztofCwalina) [inline sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs:91]: Is there a reason why we create a new classifier here as opposed to using a singleton?

**Review by KrzysztofCwalina — APPROVED** [review]

File: sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs (hunk 5 above)
Comment (annelo-msft) [reply sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs:91]: I'm planning to change this per a bug @pakrym identified.

File: sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs (hunk 1 above)
Comment (annelo-msft) [reply sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs]: Addressed.

File: sdk/core/Azure.Core/src/RequestFailedException.cs [hunk 6]
```diff
@@ -66,6 +67,19 @@ public RequestFailedException(int status, string message, string? errorCode, Exc
             ErrorCode = errorCode;
         }
 
+        internal RequestFailedException(int status, (string Message, string? ErrorCode) details):
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/RequestFailedException.cs:70]: Do we have tests for this ctor?

File: sdk/core/Azure.Core/src/ResponseClassifier.cs [hunk 7]
```diff
@@ -57,5 +60,23 @@ public virtual bool IsErrorResponse(HttpMessage message)
             var statusKind = message.Response.Status / 100;
             return statusKind == 4 || statusKind == 5;
         }
+
+        internal static (string Message, string? ErrorCode) GetErrorDetails(Response response)
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/ResponseClassifier.cs:64]: Shoud we inline this logic into ResponseFailedException for now? The ResponseFailedException is the only consumer of it and the method is static.

File: sdk/core/Azure.Core/src/RequestFailedException.cs (hunk 6 above)
Comment (annelo-msft) [reply sdk/core/Azure.Core/src/RequestFailedException.cs:70]: https://github.com/Azure/azure-sdk-for-net/issues/25554

Comment (pakrym) [reply sdk/core/Azure.Core/src/RequestFailedException.cs:70]: Let's make sure we have tests before it's time to ship.

File: sdk/core/Azure.Core/src/Response.cs [hunk 8]
```diff
@@ -82,6 +82,14 @@ public virtual BinaryData Content
...
+        /// Indicates whether the message's <see cref="ResponseClassifier"/> considers this
+        /// response an error.
+        /// </summary>
+        public bool IsError { get; internal set; }
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/Response.cs:89]: We need to switch all Core consumers to use this property, do you mind filing an issue?

File: sdk/core/Azure.Core/src/Response.cs [hunk 9]
```diff
@@ -82,6 +82,14 @@ public virtual BinaryData Content
...
+
+        internal HttpMessageSanitizer? Sanitizer { get; set; }
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/Response.cs:91]: We need to switch all Core consumers to use this property, do you mind filing an issue?

**Review by pakrym — APPROVED** [review]

File: sdk/core/Azure.Core/src/Response.cs (hunk 8 above)
Comment (annelo-msft) [reply sdk/core/Azure.Core/src/Response.cs:89]: https://github.com/Azure/azure-sdk-for-net/issues/25555

File: sdk/core/Azure.Core/src/Response.cs (hunk 9 above)
Comment (annelo-msft) [reply sdk/core/Azure.Core/src/Response.cs:91]: https://github.com/Azure/azure-sdk-for-net/issues/25555

**Pull request merged**